pkgs/development/libraries/science/math/nccl/default.nix

```diff
@@ -1,46 +1,43 @@
-{ lib
+{ autoAddOpenGLRunpathHook
 , backendStdenv
+, cuda_cccl
+, cuda_cudart
+, cuda_nvcc
+, cudaFlags
+, cudaVersion
 , fetchFromGitHub
+, lib
 , which
-, cudaPackages ? { }
-, addOpenGLRunpath
 }:
-
-with cudaPackages;
-
-let
-  # Output looks like "-gencode=arch=compute_86,code=sm_86 -gencode=arch=compute_86,code=compute_86"
-  gencode = lib.concatStringsSep " " cudaFlags.gencode;
-in
-backendStdenv.mkDerivation rec {
-  name = "nccl-${version}-cuda-${cudaPackages.cudaMajorVersion}";
+backendStdenv.mkDerivation (finalAttrs: {
+  name = "nccl-${finalAttrs.version}-cuda-${cudaVersion}";
   version = "2.16.5-1";
 
   src = fetchFromGitHub {
     owner = "NVIDIA";
     repo = "nccl";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     hash = "sha256-JyhhYKSVIqUKIbC1rCJozPT1IrIyRLGrTjdPjJqsYaU=";
   };
 
   outputs = [ "out" "dev" ];
 
   nativeBuildInputs = [
-    which
-    addOpenGLRunpath
+    autoAddOpenGLRunpathHook
     cuda_nvcc
+    which
   ];
 
   buildInputs = [
-    cuda_cudart
-  ] ++ lib.optionals (lib.versionAtLeast cudaVersion "12.0.0") [
-    cuda_cccl
+    cuda_cccl #include <nv/target>
```

Comment:
previously `cuda_cccl` was restricted to cudaVersion>=12. is there a reason to add it to builds for all versions?

Reply:
When I had tested it I needed it on all versions of CUDA iirc. I can test again later but I remember getting errors until I did that.

Reply:
@connorbaker reminder to test when I get a chance

Reply:
It was, in fact, required only for CUDA 12.0 and later.

The error I was running into was actually a bug due to the comparison with a version number with three components -- `versionAtLeast "12.0" "12.0.0"` evaluates to `false` so I was seeing build failures with `cudaPackages_12_0`.